File: deamhome/src/main/java/sosteam/deamhome/domain/item/entity/Item.java
package sosteam.deamhome.domain.item.entity;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import sosteam.deamhome.domain.account.entity.Account;
import sosteam.deamhome.domain.category.entity.ItemCategory;
import sosteam.deamhome.domain.category.entity.ItemDetailCategory;
import sosteam.deamhome.global.entity.Image;
import sosteam.deamhome.global.entity.LogEntity;

import java.util.List;

@Document
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Item extends LogEntity {

    @NotNull
    private String title;

    private String content;

    private String summary;

    private int price = 0;

    private int sellCnt = 0;

    private int wishCnt = 0;

    private int clickCnt = 0;

    private Double avgReview = 0.0;

    private int reviewCnt = 0;

    private int qnaCnt = 0;
    
    private boolean status;

    @DBRef(lazy = true)
    @Setter
    private Account account;

    @DBRef(lazy = true)
    private List<Image> images;

    @DBRef(lazy = true)
    @Setter
    private ItemCategory itemCategory;

    @DBRef(lazy = true)
    @Setter
    private ItemDetailCategory itemDetailCategory;

    public List<Image> addImage(Image image) {
        images.add(image);

        return images;
    }
}

File: src/main/kotlin/sosteam/deamhome/domain/item/entity/Item.kt
package sosteam.deamhome.domain.item.entity

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import sosteam.deamhome.global.attribute.ItemStatus
import sosteam.deamhome.global.entity.BaseEntity
import java.time.OffsetDateTime

@Table("item")
data class Item(
	@Id
	@Column("id")
	var id: Long?,
	// unique column
	@Column("public_id")
	var publicId: String,
	@Column("category_public_id")
	var categoryPublicId: String,
	@Column("title")
	var title: String,
	@Column("content")
	var content: String,
	@Column("summary")
	var summary: String,
	@Column("price")
	var price: Int = 0,
	@Column("sell_cnt")
	var sellCnt: Int = 0,
	@Column("wish_cnt")
	var wishCnt: Int = 0,
	@Column("click_cnt")
	var clickCnt: Int = 0,
	@Column("stock_cnt")
	var stockCnt: Int = 0,
	@Column("avg_review")
	var avgReview: Double = 0.0,
	@Column("review_cnt")
	var reviewCnt: Int = 0,
	// 리뷰 점수 1점 ~ 5점
	@Column("review_score")
	var reviewScore: List<Int> = listOf(0, 0, 0, 0, 0),
	@Column("qna_cnt")
	var qnaCnt: Int = 0,
	@Column("status")
	var status: ItemStatus,
	@Column("store_id")
	var storeId: String,
	@Column("free_delivery")
	var freeDelivery: Boolean = false,
	// 옵션 선택
	@Column("option")
	var option: List<String> = listOf(),
	// 상품번호
	@Column("product_number")
	var productNumber: String,
	// 마감일
	@Column("deadline")
	var deadline: OffsetDateTime,
	// 원작
	@Column("original_work")
	var originalWork: String,
	// 재질
	@Column("material")
	var material: String,
	// 크기
	@Column("size")
	var size: String,
	// 무게
	@Column("weight")
	var weight: String,
	// 배송비
	@Column("shipping_cost")
	var shippingCost: Int,
	@Column("image_urls")
	var imageUrls: List<String> = listOf()
) : BaseEntity() {
}
